src/auth/awsclient.ts

```diff
@@ -88,13 +88,19 @@ export class AwsClient extends BaseExternalAccountClient {
     // This is only required if the AWS region is not available in the
     // AWS_REGION or AWS_DEFAULT_REGION environment variables.
     this.regionUrl = options.credential_source.region_url;
+    this.validateMetadataServerUrlIfAny(this.regionUrl, 'region_url');
```

Comment:
Can you group validation of the AWS endpoints to one method that is called once? e.g. validateAwsEndpoints()